src/Sarif/Writers/CachingLogger.cs

```diff
@@ -16,6 +17,7 @@ public class CachingLogger : BaseLogger, IAnalysisLogger
     {
         public CachingLogger(IEnumerable<FailureLevel> levels, IEnumerable<ResultKind> kinds) : base(levels, kinds)
         {
+            s_rwl = new ReaderWriterLock();
```

Comment:
>s_rwl [](http://example.com/codeflow?start=12&length=5)

this is not a good name, please avoid acronyms/abbreviations and names that simply express the type of something. 

Generally, it is more useful to encode the purpose of something rather than the mechanism.